akka-actor/src/main/scala/akka/dispatch/VirtualThreadConfigurator.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2009-2025 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.dispatch
+
+import akka.annotation.InternalApi
+import com.typesafe.config.Config
+
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+import java.util.concurrent.ThreadFactory
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[akka] object VirtualThreadConfigurator {
+
+  // Note: since we still support JDK 11 and 17, we need to access the factory method reflectively
+  private def virtualThreadFactory(virtualThreadName: String): ThreadFactory = {
+    val ofVirtualMethod =
+      try {
+        classOf[Thread].getMethod("ofVirtual")
+      } catch {
+        case _: NoSuchMethodError =>
+          throw new IllegalStateException("Virtual thread executors only supported on JDK 21 and newer")
+      }
+    val ofVirtual = ofVirtualMethod.invoke(null)
+    // java.lang.ThreadBuilders.VirtualThreadBuilder is package private
```

Comment:
How would you write this without reflection?

Reply:
isn't the public way like this?
```
ThreadFactory virtualThreadFactory = Thread.ofVirtual().factory();
```

Reply:
Yep, that's what we are calling here, with a potential name inbetween `Thread.ofVirtual().name(virtualThreadName).factory()`

Reply:
To clarify the comment, the reason why we can't just do `ofVirtual.getClass.getMethod("name")` is that it is of a package private type, so we need to look up the interface it implements that has the `name` method instead.

Reply:
I was thinking we could use `Thread.Builder.OfVirtual`, which is the public type. I think this should work:
```
val ofVirtual = ofVirtualMethod.invoke(null)
val nameMethod = ofVirtual.getClass.getMethod("name", classOf[String])
nameMethod.invoke(ofVirtual, virtualThreadName)
```

Reply:
`Thread.Builder.OfVirtual` only exists on JDK 21 and later so we can't do `classOf[Thread.Builder.OfVirtual]` and have the code still compile on 11 and 17

Reply:
sorry, I above comment about `ThreadBuilders.VirtualThreadBuilder` was misleading me. I thought you reached out for that one. You are already using the one that I suggested.
```
val ofVirtualInterface = dynamicAccess.getClassFor[AnyRef]("java.lang.Thread$Builder$OfVirtual").get
```

Reply:
Ah, I'll drop the comment, it's not needed and worse if it is confusing :)